### ISSUE TITLE
Adding Grid Background Feature like Math Notebook Background

### ISSUE THREAD

**Issue by TUNA-NOPE** [excalidraw/excalidraw#6773]
Hi Excalidraw team,

I hope this message finds you well. I have been using Excalidraw for my mathematical illustrations and diagrams, and I find it to be an excellent tool. However, I believe it would greatly enhance the user experience if there was a grid lines feature that resembles a math notebook background.

The grid lines would provide a visual aid for aligning elements accurately, making it easier to create math-related diagrams, graphs, and geometric shapes. The math notebook background would add a touch of familiarity and make the workspace feel more authentic for mathematical illustrations.

I understand that Excalidraw is designed to be a versatile and intuitive tool, and the addition of grid lines with a math notebook background would further enhance its usefulness for math-related tasks. I believe many users, including students, teachers, and professionals working in mathematics and related fields, would greatly benefit from this feature.

Thank you for considering my request. I appreciate the hard work and effort you put into developing and improving Excalidraw. I look forward to seeing the continued growth of this fantastic tool.

### Feature Details
- When enabled, the grid background should overlay the canvas and provide a visual grid pattern.
- Users should have control over the grid size, spacing, and color.
- The grid should be customizable to support different types of grids (e.g., square, isometric, etc.).
- It would be helpful to have an option to toggle the grid on/off based on user preference.

### Expected Behavior
1. When the grid background feature is enabled, a grid pattern should be displayed on the canvas.
2. The grid pattern should follow the specified grid size, spacing, and color preferences.
3. The grid should not interfere with existing drawing functionality and should not obstruct or affect user interaction with the canvas.

Thank again for considering this feature request. I believe it would enhance the usability and flexibility of Excalidraw for many users. Please let me know if you need any further information or clarification.

**Comment by wloske** [comment]
Hi @TUNA-NOPE ...

I wanted to open an issue on isometric grid alone when I found this feature request.

I think you are not aware that there actually is a grid. Right cllick on the canvas and activate "Grid Mode". There's also a keyboard shortcut `CTRL+'` (which is unusable for me on my .de keyboard layout).

What I would greatly appreciate having is an isometric grid. It would help a LOT doing sketches of anything 3D ...

**Comment by dimateos** [comment]
Yeah I thought I could not use `CTRL+'` either but turns out I can with `CTRL+SHIFT+...` @wloske

![image](https://github.com/excalidraw/excalidraw/assets/26868611/6502f81b-aed6-4a97-ba96-643fcdb839c4)

* But I believe that the grid is very important, enough to at least have a dedicated button; or two, one for visuals and one for enabling snapping. Probably at the bottom left next to the zoom level etc. 

Also, along alternative textures, many programs have variable scale for grids. I mainly want grids for visual reference as I request in https://github.com/excalidraw/excalidraw/issues/6889, but otherwise snapping would have to be adjusted for the scale and potential isometric grids. Letting the user upload a texture file for full flexibility would break snapping for sure.

* This controls at the bottom left could be a prototype:

![image](https://github.com/excalidraw/excalidraw/assets/26868611/fad83e1f-3b2e-4be4-ae2c-86e78a1b4afa)

* These are some common backgrouns used in drawing apps: I personally like the current one and the dots. Probably the isometric ones should have some dashed lines or faded/bold parts.

![image](https://github.com/excalidraw/excalidraw/assets/26868611/da66c01c-8dfc-494c-8acf-94c1e80b0d37)
> Photosop background like suggested in https://github.com/excalidraw/excalidraw/issues/5732
![image](https://github.com/excalidraw/excalidraw/assets/26868611/207f1a12-a1f0-4c9d-94b7-512b192922b0)

* Note that there are some libraries for drawing math that include some grids with axis etc:
    * But this interferes with the use of the canvas and selecting items (like @TUNA-NOPE said: "The grid should not interfere with existing drawing functionality and should not obstruct or affect user interaction with the canvas.")

![image](https://github.com/excalidraw/excalidraw/assets/26868611/4c8ba573-651b-41ba-a91a-5fd31a1d296c)

**Comment by mohammadkarbalaee** [comment]
Please add this feature. for God's sake.

**Comment by MrIntellectX** [comment]
+1

**Comment by KylanZzz** [comment]
+ 2

**Comment by MaxOSchulte** [comment]
I'd love to see and use this! +1

**Comment by Amooryjubran** [comment]
Please add this feature

**Comment by STOI123** [comment]
please add this feature

**Comment by regerohan** [comment]
I would really like this feature.

**Comment by SamuLng** [comment]
+1

**Comment by alex-titarenko** [comment]
+1

**Comment by IamTheRichie** [comment]
+1

**Comment by Y-askour** [comment]
+1

**Comment by ShyneTurtle** [comment]
+1

**Comment by Mack127444** [comment]
+1

**Comment by maximecharriere** [comment]
+1

**Comment by maximecharriere** [comment]
@dwelle I think this issue can be cloased. These PR added the requested grid feature :
- #1788
- #8364

**Comment by nitishpatel** [comment]
Created a PR which does something similar and gives a notebook like feel:
https://github.com/excalidraw/excalidraw/pull/9209

**Comment by bhargavmodak** [comment]
[Option for Isometric Grid](https://github.com/excalidraw/excalidraw/discussions/5345)

I made this discussion too, a while ago, regarding adding isometric grids to excalidraw

It would be so useful to have the option to change the grid layout.

**Comment by potatochipcoconut** [comment]
this is honestly the missing feature from excalidraw... imo should be top priority to add grid lines

**Comment by jollaWellbuur** [comment]
this looks very promising:
https://github.com/excalidraw/excalidraw/pull/9893
